### PR TITLE
Change ToStringVisitor to not specify type on RecursiveVisitor to fac…

### PR DESCRIPTION
**Pull request by jensjoha**
…ilitate change on dart side

## Description

This is needed to facilitate https://dart-review.googlesource.com/c/sdk/+/172180.

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ analyzer:
   ]
   strong-mode:
     implicit-casts: false
-    implicit-dynamic: false
+    # implicit-dynamic: false # at least temporary for RecursiveVisitor change
```
Comment (dnfield) [inline analysis_options.yaml:24]: I'd like to avoid doing this - can we just opt-in the line specifically?

Comment (dnfield) [inline analysis_options.yaml:24]: Is there any way to just opt-out the line or file specifically?

We should also file a TODO to uncomment this when the SDK rolls.

**Review by dnfield — COMMENTED** [review]
LGTM with nit about TODO

File: analysis_options.yaml (hunk 1 above)
Comment (jensjoha) [reply analysis_options.yaml]: TODO added. I can't make opting out only the line work :(

Comment (jensjoha) [reply analysis_options.yaml]: TODO added. I can't make opting out only the line work :(

**Comment by dnfield** [comment]
Instead of this, should we just finish your CL to move this to the Dart repo and remove it from here?

**Comment by jensjoha** [comment]
I assume it's @johnniwinther 's CL --- but sounds good. I'll close this PR.

**Comment by dnfield** [comment]
Ahh sorry I missed that it was two different people working on this

**Pull request closed without merging**